src/SubscribeMe/Subscriber/MailchimpSubscriber.php

```diff
@@ -88,23 +96,21 @@ public function subscribe(string $email, array $options, array $userConsents = [
         }
 
         try {
-            $res = $this->getClient()->request('POST', $uri, [
-                'http_errors' => true,
-                'auth' => [$this->getApiKey(), $this->getApiSecret()],
-                'headers' => [
-                    'Content-Type' => 'application/json',
-                ],
-                'body' => json_encode($body)
-            ]);
-
-            if ($res->getStatusCode() === 200 ||  $res->getStatusCode() === 201) {
-                /** @var array $body */
-                $body = json_decode($res->getBody()->getContents(), true);
-                return $body['id'];
+            if (!is_string(json_encode($body))) {
+                throw new \InvalidArgumentException('Body missing');
             }
-        } catch (ClientException $exception) {
-            $res = $exception->getResponse();
-            if ($res->getStatusCode() === 400) {
+            $bodyStreamed = $this->getStreamFactory()->createStream(json_encode($body));
```

Comment:
You are encoding json twice.

```suggestion
            $bodyStreamed = $this->getStreamFactory()->createStream(json_encode($body, JSON_THROW_ON_ERROR));
```